src/python/pants/engine/internals/parser_test.py

```diff
@@ -17,18 +18,70 @@ def test_imports_banned() -> None:
     assert "Import used in dir/BUILD at line 4" in str(exc.value)
 
 
-def test_unrecognized_symbol() -> None:
-    parser = Parser(
-        target_type_aliases=["tgt"],
-        object_aliases=BuildFileAliases(
-            objects={"obj": 0},
-            context_aware_object_factories={"caof": lambda parse_context: lambda _: None},
-        ),
+def test_unrecogonized_symbol() -> None:
+    def perform_test(tta: list, good_err: str) -> None:
+        parser = Parser(
+            target_type_aliases=tta,
+            object_aliases=BuildFileAliases(
+                objects={"obj": 0},
+                context_aware_object_factories={"caof": lambda parse_context: lambda _: None},
+            ),
+        )
+        prelude_symbols = BuildFilePreludeSymbols(FrozenDict({"prelude": 0}))
+        with pytest.raises(ParseError) as exc:
+            parser.parse("dir/BUILD", "fake", prelude_symbols)
+        assert (
+            str(exc.value)
+            == good_err
```

Comment:
There's a lot of duplication below in having to repeat the entire expected error message every time. It looks like the only parts that are changing is the did you mean snippet + the "all registered symbols". 

You can duplicate this by dynamically creating the `expected` message here in this helper. Instead of taking `good_err: str`, for example, take `did_you_mean: str` and then the above suggestion of changing `tta` to `extra_targets: list[str]`. Then, use an f string to insert those into the expected message.